pkg/types/config.go

```diff
@@ -44,10 +44,14 @@ func (c SecretConfig) BuildURL(path string) (string, error) {
 	return buildURL(c.Protocol, c.Host, path, c.Port)
 }
 
-// BuildSecretsPathURL constructs a URL which can be used to identify a secret's path
-// subPath is the location of the secrets in the secrets engine
-func (c SecretConfig) BuildSecretsPathURL(subPath string) (string, error) {
-	return c.BuildURL(c.Path + subPath)
+// BuildSecretsPathURL constructs a URL to the service's secret with in it's secret store
+// secretName is the name of the secret in the service's secret store
+func (c SecretConfig) BuildSecretsPathURL(secretName string) (string, error) {
+	return c.BuildURL(fmt.Sprintf("%s/%s", c.StoreBasePath, secretName))
+}
+
+func (c SecretConfig) BuildRequestPathURL(subPath string) (string, error) {
```

Comment:
Also need to add comment for the function

```suggestion
func (c SecretConfig) BuildRequestURL(subPath string) (string, error) {
```